Possibly flaky test test_that_the_plot_window_contains_the_expected_elements

When running "Python coverage" GA workflow, Test GUI part, test is failing with the following error:
```
FAILED tests/unit_tests/gui/test_main_window.py::test_that_the_plot_window_contains_the_expected_elements - AssertionError: assert ['default', 'default_0'] == ['default', '..., 'default_3']
  Right contains 3 more items, first extra item: 'default_1'
  Full diff:
  - ['default', 'default_0', 'default_1', 'default_2', 'default_3']
  + ['default', 'default_0']
```

Could not reproduce the error when running on my machine

--- a/src/ert/gui/tools/plot/plot_case_selection_widget.py
+++ b/src/ert/gui/tools/plot/plot_case_selection_widget.py
@@ -69,6 +69,7 @@ class CaseSelectionWidget(QWidget):
         widget.setLayout(layout)
 
         combo = QComboBox()
+        combo.setObjectName("case_selector")
         combo.setSizeAdjustPolicy(QComboBox.AdjustToMinimumContentsLengthWithIcon)
         combo.setMinimumContentsLength(20)
         combo.setModel(self.__model)

--- a/tests/unit_tests/gui/conftest.py
+++ b/tests/unit_tests/gui/conftest.py
@@ -221,26 +221,27 @@ def ensemble_experiment_has_run(opened_main_window, run_experiment, request):
     with open("poly_eval.py", "w", encoding="utf-8") as f:
         f.write(
             dedent(
-                """#!/usr/bin/env python
-import numpy as np
-import sys
-import json
+                """\
+                #!/usr/bin/env python
+                import numpy as np
+                import sys
+                import json
 
-def _load_coeffs(filename):
-    with open(filename, encoding="utf-8") as f:
-        return json.load(f)
+                def _load_coeffs(filename):
+                    with open(filename, encoding="utf-8") as f:
+                        return json.load(f)
 
-def _evaluate(coeffs, x):
-    return coeffs["a"] * x**2 + coeffs["b"] * x + coeffs["c"]
+                def _evaluate(coeffs, x):
+                    return coeffs["a"] * x**2 + coeffs["b"] * x + coeffs["c"]
 
-if __name__ == "__main__":
-    if np.random.random(1) > 0.5:
-        sys.exit(1)
-    coeffs = _load_coeffs("coeffs.json")
-    output = [_evaluate(coeffs, x) for x in range(10)]
-    with open("poly_0.out", "w", encoding="utf-8") as f:
-        f.write("\\n".join(map(str, output)))
-        """
+                if __name__ == "__main__":
+                    if np.random.random(1) > 0.5:
+                        sys.exit(1)
+                    coeffs = _load_coeffs("coeffs.json")
+                    output = [_evaluate(coeffs, x) for x in range(10)]
+                    with open("poly_0.out", "w", encoding="utf-8") as f:
+                        f.write("\\n".join(map(str, output)))
+                """
             )
         )
     os.chmod(

--- a/tests/unit_tests/gui/conftest.py
+++ b/tests/unit_tests/gui/conftest.py
@@ -1,3 +1,4 @@
+import contextlib
 import copy
 import fileinput
 import os.path
@@ -139,10 +140,8 @@ def run_experiment(request, opened_main_window):
     def func(experiment_mode):
         qtbot = QtBot(request)
         gui = opened_main_window
-        try:
+        with contextlib.suppress(FileNotFoundError):
             shutil.rmtree("poly_out")
-        except FileNotFoundError:
-            pass
         # Select correct experiment in the simulation panel
         simulation_panel = gui.findChild(SimulationPanel)
         assert isinstance(simulation_panel, SimulationPanel)

--- a/tests/unit_tests/gui/conftest.py
+++ b/tests/unit_tests/gui/conftest.py
@@ -178,8 +178,8 @@ def run_experiment(request, opened_main_window):
 
             qtbot.mouseClick(run_dialog.done_button, Qt.LeftButton)
 
-        QTimer.singleShot(1000, use_rundialog)
         qtbot.mouseClick(start_simulation, Qt.LeftButton)
+        use_rundialog()
 
     return func
 

--- a/tests/unit_tests/gui/test_main_window.py
+++ b/tests/unit_tests/gui/test_main_window.py
@@ -1,5 +1,5 @@
 import os
-from typing import List
+from typing import Optional
 
 import pytest
 from qtpy.QtCore import Qt, QTimer
@@ -37,6 +37,13 @@ def test_that_the_plot_window_contains_the_expected_elements(
     esmda_has_run, opened_main_window, qtbot
 ):
     gui = opened_main_window
+    expected_cases = [
+        "default",
+        "default_0",
+        "default_1",
+        "default_2",
+        "default_3",
+    ]
 
     # Click on Create plot after esmda has run
     plot_tool = gui.tools["Create plot"]
@@ -50,24 +57,16 @@ def test_that_the_plot_window_contains_the_expected_elements(
     case_selection = plot_window.findChild(CaseSelectionWidget)
     data_types = plot_window.findChild(DataTypeKeysWidget)
     assert isinstance(data_types, DataTypeKeysWidget)
-    combo_boxes: List[QComboBox] = case_selection.findChildren(
-        QComboBox
+    combo_box: Optional[QComboBox] = case_selection.findChild(
+        QComboBox, "case_selector"
     )  # type: ignore
+    assert combo_box is not None
 
     # Assert that the Case selection widget contains the expected cases
     case_names = []
-    assert len(combo_boxes) == 1
-    combo_box = combo_boxes[0]
     for i in range(combo_box.count()):
-        combo_box.setCurrentIndex(i)
-        case_names.append(combo_box.currentText())
-    assert sorted(case_names) == [
-        "default",
-        "default_0",
-        "default_1",
-        "default_2",
-        "default_3",
-    ]
+        case_names.append(combo_box.itemText(i))
+    assert sorted(case_names) == expected_cases
 
     data_names = []
     data_keys = data_types.data_type_keys_widget
@@ -93,25 +92,21 @@ def test_that_the_plot_window_contains_the_expected_elements(
         "Statistics",
     }
 
-    # Add all the cases
-    for name in case_names:
-        combo_box = combo_boxes[-1]
-        for i in range(combo_box.count()):
-            if combo_box.itemText(i) == name:
-                combo_box.setCurrentIndex(i)
+    # Add cases to plot
+    for _ in expected_cases:
         qtbot.mouseClick(
             case_selection.findChild(QToolButton, name="add_case_button"), Qt.LeftButton
         )
-        combo_boxes: List[QComboBox] = case_selection.findChildren(
-            QComboBox
-        )  # type: ignore
-    assert {x.currentText() for x in case_selection.findChildren(QComboBox)} == {
-        "default",
-        "default_0",
-        "default_1",
-        "default_2",
-        "default_3",
-    }
+    all_added_combo_boxes = case_selection.findChildren(QComboBox, name="case_selector")
+    assert len(expected_cases) == len(all_added_combo_boxes)
+
+    # make sure pairwise different and all cases are selected
+    for case_index, case_name in enumerate(expected_cases):
+        combo_box = all_added_combo_boxes[case_index]
+        for i in range(combo_box.count()):
+            if case_name == combo_box.itemText(i):
+                combo_box.setCurrentIndex(i)
+    assert {box.currentText() for box in all_added_combo_boxes} == set(expected_cases)
 
     # Cycle through showing all the tabs and plot each data key
 

